src/rust/ensogl/lib/components/src/list_view/entry.rs

```diff
@@ -350,156 +265,112 @@ impl List {
-    pub fn update_entries(&self, mut range:Range<Id>) {
-        range.end = range.end.min(self.provider.get().entry_count());
-        if range != self.entries_range.get() {
-            debug!(self.logger, "Update entries for {range:?}");
-            let provider = self.provider.get();
-            let current_entries:HashSet<Id> = with(self.entries.borrow_mut(), |mut entries| {
-                entries.resize_with(range.len(),|| self.create_new_entry());
-                entries.iter().filter_map(|entry| entry.id.get()).collect()
-            });
-            let missing = range.clone().filter(|id| !current_entries.contains(id));
-            // The provider is provided by user, so we should not keep any borrow when calling its
-            // methods.
-            let models = missing.map(|id| (id,provider.get(id)));
-            with(self.entries.borrow(), |entries| {
-                let is_outdated = |e:&Entry| e.id.get().map_or(true, |i| !range.contains(&i));
-                let outdated    = entries.iter().filter(|e| is_outdated(e));
-                for (entry,(id,model)) in outdated.zip(models) {
-                    Self::update_entry(&self.logger,entry,id,&model);
+    fn update_visible_entries(&self) {
+        let entry_at_y_saturating = |y:f32| {
+            match Self::entry_at_y_position(y,self.entry_count()) {
+                IdAtYPosition::AboveFirst => 0,
+                IdAtYPosition::UnderLast  => self.entry_count().saturating_sub(1),
+                IdAtYPosition::Entry(id)  => id,
+            }
+        };
+        let first_visible          = entry_at_y_saturating(self.visible_range.get().end);
+        let last_visible           = entry_at_y_saturating(self.visible_range.get().start);
+        let visible_ids: Range<Id> = first_visible..(last_visible +1);
+
+        // It can be extremely slow to create or destroy objects, in particular text areas.
+        // Therefore, we only destroy or create those that enter or leave the visible area.
+
+        {  // Remove entries that went out of sight
+            let mut visible_entries = self.visible_entries.borrow_mut();
+
+            let removed_entries = visible_entries.drain_filter(|id, _| !visible_ids.contains(id));
+            for (_, entry) in removed_entries {
+                self.display_object.remove_child(&entry);
+            }
+        }
+
+        // Add entries that came into sight
+        for id in visible_ids {
+            let needs_to_be_created = !self.visible_entries.borrow().contains_key(&id);
+            if needs_to_be_created {
+                if let Some(new_entry) = self.provider.get().get(&self.app,id) {
+                    self.display_object.add_child(&new_entry);
+                    self.visible_entries.borrow_mut().insert(id, new_entry);
+                    self.config_entry(id);
                 }
-            });
-            self.entries_range.set(range);
+            }
+        }
```

Comment:
I compared the performance when I made this change and did not see a significant difference. But the running time is dominated by the time that it takes to set the text on text areas, which takes much longer than it should.

Reply:
"I compared the performance when I made this change and did not see a significant difference" - Imo this is not a correct explanation. Until you made extensive benchmarks this does not explain anything. 

"But the running time is dominated by the time that it takes to set the text on text areas" – this doesn't matter at all. When text areas will be improved, this PR introduces slowdown, it should be reverted. I agree with Adam, before making such changes, a design call / discussion should be made. For now, please revert to previous, faster solution.

Reply:
The simple reverting is not possible: we can keep it or rewrite as I described.

The thing is, that before the providers provided _models_ only with icon and text. But now we want support for anything (e.g. icon + text + icon).

Reply:
What were the reasons to choose the previous implementation. Where there any benchmarks? Or is there a reason to assume that the new one is significantly slower?

Reply:
Never said it is _significantly_ slower for sure. I only asked to make sure it isn't. 

There were no benchmarks for the previous solution. Actually, it was picked, because I just implemented it that way (it wasn't so hard after all). I checked the scrolling performance on list_view debug scene, however I think that was in times when setting text in text_area was not so slow.

Reply:
I created an issue for it (#1701).